package.json

```diff
@@ -21,6 +21,11 @@
       "packages/*",
       "packages/@unimodules/*",
       "react-native-lab/react-native"
+    ],
+    "nohoist": [
+      "**/@storybook/**",
```

Comment:
nohoist is pretty fragile on top of Yarn's fragility. I intentionally made expo-yarn-workspaces work without it. If this causes issues with the rest of the repo one day we'll have to remove this.

Reply:
storybook uses a lot of node require things to look up the file-system and doesn't account for workspaces. @shilman any thoughts

Reply:
@EvanBacon No specific support for workspaces & have seen various workarounds in our issues with no real killer issues AFAIK. If you want to spec what first class workspace support might look like, we might be able to get it built as part of 5.x. I believe those `require.resolve` are an @ndelangen signature move, so he'd be the best person to comment on all that.

Reply:
`require` and `require.resolve` is usually a solution to workspace issues, since it actually walks up the tree to find things in `node_modules` above it, rather than hardcoding `node_modules/thing`, where `thing` might be hoisted up. Using `require.resolve('./jest/setup')` in the Jest preset instead of `<rootDir>/node_modules/react-native/jest/setup` fixes a bunch of issues with jest and react-native in monorepos for instance.

I don't know what the issue with storybook is here, but maybe they're missing a dependency declaration, so they load a version from some other dependency? Or vice versa, where some expo module (or one of its dep) is missing a dependency on a module it uses, one from `storybook` is hoisted, and the wrong one is loaded?

---

We had an issue at work where we used Emotion 10 and `react-select` used Emotion 9. We ended up in a situation where emotion 9 was hoisted to the top, and we got 2 version of emotion 10 in our bundle, with separate caches, which resulted in some super weird styling bugs. The solution was to add a dependency on `emotion@10` to the root `package.json`, so we forced the correct version to be hoisted.

Reply:
what happens if you hoist the packages from storybook? what breaks?